vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -303,25 +303,27 @@ and
     member this.SyncProject(project: AbstractProject, projectContext: IWorkspaceProjectContext, site: IProjectSite, forceUpdate) =
         let hashSetIgnoreCase x = new HashSet<string>(x, StringComparer.OrdinalIgnoreCase)
         let updatedFiles = site.SourceFilesOnDisk() |> hashSetIgnoreCase
-        let workspaceFiles = project.GetCurrentDocuments() |> Seq.map(fun file -> file.FilePath) |> hashSetIgnoreCase
+        let workspaceFiles = project.GetCurrentDocuments() |> Seq.map (fun file -> file.FilePath) |> hashSetIgnoreCase
         
         let mutable updated = forceUpdate
 
         for file in updatedFiles do
             if not(workspaceFiles.Contains(file)) then
                 projectContext.AddSourceFile(file)
                 updated <- true
+        
         for file in workspaceFiles do
             if not(updatedFiles.Contains(file)) then
                 projectContext.RemoveSourceFile(file)
                 updated <- true
         
-        let updatedRefs = site.AssemblyReferences() |> hashSetIgnoreCase
+        let updatedRefs = site.AssemblyReferences() |> Seq.filter File.Exists |> hashSetIgnoreCase
```

Comment:
Is this correct?  For cross-project references the assembly reference should still be added to the project, shouldn't it, even if the assembly doesn't exist on disk yet?  Or should we be adding cross-project references to the Roslyn project differently?